<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SettingsWindow</class>
 <widget class="QWidget" name="SettingsWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Settings</string>
  </property>
  <layout class="QVBoxLayout" name="mainLayout">
   <item>
    <widget class="QGroupBox" name="apiSettingsGroup">
     <property name="title">
      <string>API Settings</string>
     </property>
     <layout class="QFormLayout" name="apiFormLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="apiTokenLabel">
        <property name="text">
         <string>API Token:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="apiTokenInput">
        <property name="placeholderText">
         <string>Enter your API token</string>
        </property>
        <property name="minimumWidth">
         <number>400</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="embeddingModelLabel">
        <property name="text">
         <string>Embedding Model:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="embeddingModelInput">
        <property name="placeholderText">
         <string>Enter embedding model name</string>
        </property>
        <property name="minimumWidth">
         <number>400</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="llmModelLabel">
        <property name="text">
         <string>LLM Model:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLineEdit" name="llmModelInput">
        <property name="placeholderText">
         <string>Enter LLM model name</string>
        </property>
        <property name="minimumWidth">
         <number>400</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="chunkSettingsGroup">
     <property name="title">
      <string>Chunk Settings</string>
     </property>
     <layout class="QFormLayout" name="chunkFormLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="chunkSizeLabel">
        <property name="text">
         <string>Chunk Size:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="chunkSizeInput">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>10000</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="chunkOverlapLabel">
        <property name="text">
         <string>Chunk Overlap:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="chunkOverlapInput">
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>10000</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="interfaceSettingsGroup">
     <property name="title">
      <string>Interface Settings</string>
     </property>
     <layout class="QFormLayout" name="interfaceFormLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="interfaceModeLabel">
        <property name="text">
         <string>Mode:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="interfaceModeDropdown">
        <property name="minimumWidth">
         <number>150</number>
        </property>
        <item>
         <property name="text">
          <string>Light</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Dark</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="storageSettingsGroup">
     <property name="title">
      <string>Storage Settings</string>
     </property>
     <layout class="QGridLayout" name="storageGridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="documentsDirLabel">
        <property name="text">
         <string>Documents Directory:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="documentsDirInput">
        <property name="placeholderText">
         <string>Select directory...</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="browseDocumentsButton">
        <property name="text">
         <string>Browse</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="chatHistoryDirLabel">
        <property name="text">
         <string>Chat History Directory:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLineEdit" name="chatHistoryDirInput">
        <property name="placeholderText">
         <string>Select directory...</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QPushButton" name="browseChatHistoryButton">
        <property name="text">
         <string>Browse</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="buttonLayout">
     <item>
      <spacer name="spacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint">
        <size>
         <width>1</width>
         <height>1</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="resetButton">
       <property name="text">
        <string>Reset</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="saveButton">
       <property name="text">
        <string>Save</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="cancelButton">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>